classes/submission/Representation.inc.php

```diff
@@ -85,6 +85,33 @@ function getSubmissionId() {
 	}
 
 	/**
+	 * Set submission revision.
+	 * @param $submissionRevision int
+	 */
+	function setSubmissionRevision($submissionRevision) {
+		$this->setData('submissionRevision', $submissionRevision);
+	}
+
+	/**
+	 * Get submission revision.
+	 * @return int
+	 */
+	function getSubmissionRevision() {
+		return $this->getData('submissionRevision');
+	}
+
+	/**
+	 * Get the current revision id for a galley from submission
+	 * @param $contextId int
+	 * @return int
+	 */
+	function getCurrentVersionId($contextId = null) {
```

Comment:
Yes, Carola and I discussed this topic as well. But if we create a new column for the latest revision - wouldn't the data be dublicate? But of course it would be much faster. So if you prefer this way, I am happy to implement the additional column with an update on the publication of the version.